docs/DeveloperGuide.md

```diff
@@ -1,33 +1,373 @@
+If it isn’t, the Command class will detect the current mode of the program and run that command string under that 
+specific mode, distributed under handleNonGeneralCommand() method. If it still doesn’t parse successfully, 
+it will be considered an invalid command and an error message will be printed out.
+
+The classes Ui, ErrorMessage, HelpMessage and MainMenu help handle the printing of various user interfaces of the 
+program. ErrorMessage, HelpMessage and MainMenu are subclasses of Ui to make use of the methods in Ui.
+
+StudyIt class will also initialize various instances of classes such as TimeTableRun, FlashCardRun, 
+ArrayList<BookmarkCategory>, ArrayList<Grade>, ArrayList<Person> and pass it to Command class to perform 
+each mode’s functionality.
+
+### **Bookmark Component**
+
+This bookmark section consists of how the bookmark feature is implemented. 
+The bookmark feature is implemented similarly to the main architecture, however, 
+in a smaller scale and a more bookmark-specific way. Figure 1 illustrates the general overview, 
+the associations and the multiplicity of the bookmark classes.
+
+![Bookmark Class Diagram](Images/BookmarkDG/BookmarkRun_Class.png)
+
+API: java.bookmarkRun
+
+The bookmark component consists of six major classes: `BookmarkRun`, `BookmarkStorage`, `BookmarkUi`, 
+`BookmarkCategory`, `BookmarkList` and `BookmarkParser`. 
+As shown in figure 1, `BookmarkRun` is 
+* The main class to be called when the bookmark mode is accessed.
+* The main class to access other bookmark classes.
+
+Given below, Figure 2, is the sequence diagram of how the classes interact with each 
+other when bookmark mode is accessed from the main function.
+
+![Bookmark Sequence Diagram](Images/BookmarkDG/sequence_bookmark.png)
+
+The bookmark component has two modes: the main bookmark mode and the category mode. 
+As shown in Figure 2, when `BookmarkRun` is called, 
+`BookmarkParser` will be called and return a `BookmarkCommand`. 
+Afterwards, `BookmarkRun` will then call `executeCommand` in `BookmarkCommand` which executes 
+the intended actions 
+Then, it will call `getCategorymode` to get the current mode the user is in. 
+
+**Bookmark Implementation**
+
+A more detailed explanation of `BookmarkCommand`, `BookmarkCategory` and `BookmarkList` will be 
+shown below. `BookmarkUi` and `BookmarkStorage` follow the same design implementations as the main 
+architecture. The below figure shows the command classes available and they are called based on the 
+`BookmarkParser` class as illustrated in Figure 2. 
+
+![BookmarkCommand ClassDiagram](Images/BookmarkDG/bookmarkCommand_Class.png)
+
+Figure 4 shows a more detailed sequence diagram of how the `BookmarkCategory`, `BookmarkCommand` 
+and `BookmarkList` interacts with each other for the scenario when the user input an `AddLinkCommand` 
+in Bookmark mode.
+
+![BookmarkCommand ClassDiagram](Images/BookmarkDG/AddCommand_sequenceDiagram.png)
+
+###Timetable Component
```

Comment:
Not rendered properly on github pages due to the missing space in between # and T

![image](https://user-images.githubusercontent.com/60433267/97667890-d8985300-1abb-11eb-949e-17ea33c99455.png)